src/pages/dataset/[dataset_id].js

```diff
@@ -82,52 +80,43 @@ export const Dataset = ({ query }) => {
     )
   }
 
+  if (loading) {
+    return (
+      <Box align="center" fill justify="center" margin={{ top: 'large' }}>
+        <Spinner />
+      </Box>
+    )
+  }
+
   return (
     <FixedContainer>
-      {loading ? (
-        <Box align="center" fill justify="center" margin={{ top: 'large' }}>
-          <Spinner />
-        </Box>
-      ) : (
-        <Box>
-          {selectedDataset?.data && (
-            <>
-              <DatasetPageHeader dataset={selectedDataset} />
+      <Box>
+        {dataset?.data && (
```

Comment:
I think we can get rid of this check as well there should only be 2 states we care about. Loading and loaded. I would think that if dataset.data is undefined then loading would be true.

Reply:
It has been cleaned up in 41e82602de98496f22dfcb909804e15277db76f3 👍